draft-ietf-masque-h3-datagram.md

```diff
@@ -335,238 +229,50 @@ Participant mode:
 : In this mode, the intermediary terminates the data stream and parses all
 Capsule Type and Capsule Length fields it receives.
 
-Each Capsule Type determines whether it is opaque or transparent to
-intermediaries in participant mode: opaque capsules are forwarded unmodified
-while transparent ones can be parsed, added, or removed by intermediaries.
-Intermediaries MAY modify the contents of the Capsule Data field of transparent
-capsule types.
-
-Unless otherwise specified, all Capsule Types are defined as opaque to
-intermediaries. Intermediaries MUST forward all received opaque CAPSULE frames
-in their unmodified entirety. Intermediaries MUST NOT send any opaque CAPSULE
-frames other than the ones it is forwarding. All Capsule Types defined in this
-document are opaque, with the exception of the datagram capsules, see
-{{datagram-capsule}}. Definitions of new Capsule Types MAY specify that the
-newly introduced type is transparent. Intermediaries MUST treat unknown Capsule
-Types as opaque.
-
-Intermediaries respect the order of opaque CAPSULE frames: if an intermediary
-receives two opaque CAPSULE frames in a given order, it MUST forward them in
-the same order.
+Capsules MUST be forwarded unmodified by intermediaries, with the exception of
+the DATAGRAM capsule; the DATAGRAM capsule MAY be parsed, added, or removed by
+intermediaries, see {{datagram-capsule}}. Definitions of new Capsule Types MAY
+specify custom intermediary processing.
```

Comment:
👍

I might amend this to include the condition that the intermediary understand the semantics of the request enough to know that capsules are in use:

```suggestion
Capsules MUST be forwarded unmodified by intermediaries, with the exception of
the DATAGRAM capsule. An intermediary that understands the request semantics 
sufficient to know that capsules are in use MAY parse, add, or remove DATAGRAM
capsules; see {{datagram-capsule}}. Definitions of new Capsule Types MAY specify
custom intermediary processing.
```

What I don't want to see is intermediaries guessing that a particular unknown `:protocol` is using capsules when they do not.  Splicing in DATAGRAM frames/capsules in that context might go badly wrong.